R/clean_data_helpers.R

```diff
@@ -66,11 +77,22 @@ scan_columns <- function(x) {
 #'                              package = "cleanepi"))
 #' )
 scan_data <- function(data) {
-  scan_result           <- data.frame(t(apply(data, 2L, scan_columns)))
-  names(scan_result)    <- c("missing", "numeric", "date", "character",
+  # when scanning through the data, logical and factor columns will be treated
+  # differently from the others. It means only the percent of missing and Date
+  # values will be evaluated for these columns.
+  # The percent of numeric and character value will be set automatically to 0 as
+  # to prevent from the effects of the conversion to numeric and character.
+  #
+  types         <- as.character(vapply(data, class, character(1L)))
```

Comment:
Do we want to use `class()` or `typeof()` here?

```suggestion
  types         <- vapply(data, class, character(1L))
```

Reply:
`class()` will gives us the inofrmation that we need - the composition in data types of every column.

`typeof()` will tell us date values are double, numeric. This is not what we want to spot out from the `scan_data()` function.

Reply:
But `class()` doesn't always return an output of length 1 so this will error.

You will see this with a test including `POSIXct` columns for example

Reply:
I still believe that `typeof()` is not appropriate for this task.

I have made changes to account for multiple types in commit `db420c8`.